DESIGN.md

```diff
@@ -54,39 +56,113 @@ Usage:
+    arguments.
+* `help`
+  * Prints the lists of database and meta commands.
+* `quit`
+  * Quits the shell.
+
+### Shell Meta Commands
 
 All commands to the shell itself will be prefixed with a backslash (\), e.g:
 
 * `\quit`
-    * Quits the shell
+  * Quits the shell. This can be preceded by the backslash or not.
 * `\help`
-    * Prints the lists of meta commands
+  * Prints the lists of meta commands. This can be preceded by the backslash or not.
 * `\status`
-    * Prints out things like connection status, server ping latency, etc.
+  * Prints out things like connection status, server ping latency, etc.
 * `\metrics [on|off|last]`
-    * Determines whether or not metrics will be emitted after the results of a query are shown.
-    * last prints out the last known metrics from your previous command.
-* `\query-results-format [ion|json|table]`
-    * ion Prints the objects from the database as ION documents in text.
-    * json Prints the objects from the database as JSON documents in text.
-    * table Tabulates the data and prints out the data as rows.
+  * Determines whether or not metrics will be emitted after the results of a query are shown.
+  * last prints out the last known metrics from your previous command.
+* `\query-results-format [ion|json|table|csv]`
+  * ion Prints the objects from the database as ION documents in text.
+  * json Prints the objects from the database as JSON documents in text.
+  * table Tabulates the data and prints out the data as rows.
+  * table Prints out the data in CSV format.
 * `\auto-commit`
-    * Determines whether each statement will be executed as a transaction or
-      not. By default this is off, meaning that statements are all executed as
-      individual transactions.
+  * Determines whether each statement will be executed as a transaction or
+    not. By default this is off, meaning that statements are all executed as
+    individual transactions.
 * `\history [limit]`
-    * Print out the last limit commands executed.
+  * Print out the last limit commands executed.
 * `\use [ledger|endpoint] ledger-or-endpoint`
-    * Connect to a different endpoint or ledger.
-* `\ledger [list|create|delete] [ledger-name]?`
-    * Show, create, or delete a ledger.
-* `\show-tables`
-    * Provide a list of tables in the current ledger.
+  * Connect to a different endpoint or ledger.
 * `\delimiter delimiter-characters`
-    * Specify the delimiter for end of command processing, such as `\n` or `;`.
+  * Specify the delimiter for end of command processing, such as `\n` or `;`.
 * `\output [console|file] [outfile]?`
-    * Write results either into a file which you specify, or into the console.
+  * Write results either into a file which you specify, or into the console.
+
+### Line terminators
+
+The way to indicate to the shell that you are done with a command is by using
+CTRL-LF (`\n` or `\r\n`). Some SQL data miners using shells regularly type out
```

Comment:
We need to flesh this out more. The *actual* way to do this right now is to send the "key press RETURN event". It's not actually `\n`. For example, if you had this:

```
qldb> select * from foo{cursor is here}
```

and you keypress ENTER, then the command is sent. But, if you opened a text editor and copied a newline and pasted it in:

```
qldb> select * from foo
{cursor is here}
```

I don't know what this means for the scripted mode.

Reply:
Have specified ENTER now. Agree it needs further work.